model/task_lifecycle.go

```diff
@@ -865,7 +865,7 @@ func getBuildStatus(buildTasks []task.Task) string {
 	// not started
 	noStartedTasks := true
 	for _, t := range buildTasks {
-		if !evergreen.IsUnstartedTaskStatus(t.Status) {
+		if !evergreen.IsUnstartedTaskStatus(t.Status) || t.Blocked() {
```

Comment:
I think this is only part of the fix -- what if we have only unstarted and blocked tasks? Do we already say that the build is failed? Or do we wait until the unstarted ones are finished? I think we only want to tweak the status if all tasks are finished or blocked.

Reply:
It seems from the ticket description that the bug isn't in this case but in the lower "do we mark it as started or failed" case.

Reply:
fixed